### PR TITLE
Improve Build-ContainerImage pipe handling

### PR DESCRIPTION
**Pull request by jstarks**
Stop using OS pipes to get a tar stream and just implement a simple ring
buffer (kind of like io.Pipe from Go). This improves the error handling of
Build-ContainerImage.

### CODE REVIEW

File: src/Docker.PowerShell/Docker.psd1 [hunk 1]
```diff
@@ -33,6 +33,7 @@ Copyright = 'Copyright (c) 2016 Microsoft'
 # Cmdlets to export from this module
 CmdletsToExport = @(
     'ConvertTo-ContainerImage',
+    'Copy-ContainerFile',
```
Comment (swernli) [inline src/Docker.PowerShell/Docker.psd1]: Did this sneak in from another commit?  I don't see Copy-ContainerFile implemented in this change...

Comment (jstarks) [inline src/Docker.PowerShell/Docker.psd1]: Oops

Comment (jstarks) [inline src/Docker.PowerShell/Docker.psd1]: Fixed

**Comment by swernli** [comment]
LGTM

**Pull request merged**